site/app/views/AutoGradingView.php

```diff
@@ -167,6 +167,21 @@ public function loadAutoChecks(\app\models\gradeable\GradedGradeable $graded_gra
                     ])
                 ];
             }
+            elseif (substr($file_path, strlen($file_path) - 6, 6) === ".ipynb") {
```

Comment:
You could use [`str_ends_with()`](https://www.php.net/manual/en/function.str-ends-with.php) to make this cleaner.  Alternatively, PHP has a [`pathinfo()`](https://www.php.net/manual/en/function.pathinfo.php) function specifically to deal with this case.